### ISSUE TITLE
[REVIEW]: Chromatiblock: scalable whole-genome visualization of structural differences in prokaryotes

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2451]
**Submitting author:** @mjsull (<a href="http://orcid.org/0000-0001-9076-6963">Mitchell</a>)
**Repository:** <a href="https://github.com/mjsull/chromatiblock/" target ="_blank">https://github.com/mjsull/chromatiblock/</a>
**Version:** v1.0.0
**Editor:** @will-rowe
**Reviewers:** @telatin, @rpetit3
**Archive:** <a href="https://doi.org/10.5281/zenodo.4034604" target="_blank">10.5281/zenodo.4034604</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/2f011b6ec30127ce83d29818aca256cb/status.svg)](https://joss.theoj.org/papers/2f011b6ec30127ce83d29818aca256cb)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/2f011b6ec30127ce83d29818aca256cb"><img src="https://joss.theoj.org/papers/2f011b6ec30127ce83d29818aca256cb/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/2f011b6ec30127ce83d29818aca256cb/status.svg)](https://joss.theoj.org/papers/2f011b6ec30127ce83d29818aca256cb)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/mjsull/chromatiblock/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@telatin & @rpetit3, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @will-rowe know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/mjsull/chromatiblock/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@mjsull) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [X] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [X] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [X] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/mjsull/chromatiblock/">repository url</a>?
- [X] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [X] **Contribution and authorship:** Has the submitting author (@mjsull) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [X] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [X] **Installation:** Does installation proceed as outlined in the documentation?
- [X] **Functionality:** Have the functional claims of the software been confirmed?
- [X] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [X] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [X] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [X] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [X] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [X] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [X] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [X] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @telatin, @rpetit3 it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02451/joss.02451/10.21105.joss.02451.pdf)

**Comment by will-rowe** [comment]
Hi all. I'm about to go on annual leave for a week. Please let me know if you need anything before I go - I'll pick things up when I get back otherwise.

I see @telatin has already done his review - thank you very much.

**Comment by rpetit3** [comment]
@will-rowe  Can the invitation link be resent, seems I was too slow to respond and it has expired. Can't edit the first comment at the moment. (Apologies!)

**Comment by will-rowe** [comment]
Hi @rpetit3 - that seems odd! I'll try reassigning you but I've not had this before. It does say that you are already assigned but let's try again

**Comment by will-rowe** [comment]
@whedon add @rpetit3 as reviewer

**Comment by whedon** [comment]
OK, @rpetit3 is now a reviewer

**Comment by rpetit3** [comment]
That did it thank you!

**Comment by will-rowe** [comment]
Hi @mjsull. Just want to check back in with this. Looks like we now have 2 excellent reviews completed for you. Please let us know when you have had chance to address these comments and we can take another look over the submission.

**Comment by will-rowe** [comment]
Hey @mjsull. Just checking in to see how you are doing with this submission? Cheers.

**Comment by mjsull** [comment]
Hi Will,

Sorry about missing your earlier email. Definitely very helpful, working
through the comments now. Will finish over the weekend.

Best,

Mitch

**Comment by will-rowe** [comment]
Hi @mjsull - just checking in to see how things are going with your submission?

**Comment by will-rowe** [comment]
Hi again @mjsull. Just want to see how you are getting on? Let me know if you need anything or if want to pause your submission?

**Comment by mjsull** [comment]
Hi Will,

Apologies, I was waiting to get feedback from my co-author (who has been a
bit snowed under by sars-cov-2 related work). He said he'd get back to me
later today.

Thanks again,

Mitch

**Comment by will-rowe** [comment]
No problem at all!

**Comment by mjsull** [comment]
Hi @will-rowe 

I think I've addressed everything. Please find my comments for each reviewer in their issue threads.

https://github.com/mjsull/chromatiblock/issues/10
https://github.com/mjsull/chromatiblock/issues/12

Best,

Mitch

**Comment by rpetit3** [comment]
@mjsull everything looks great! I have merged your latest release into Bioconda (https://github.com/bioconda/bioconda-recipes/pull/24287). Once its synced to Anaconda's CDN, I'll give it a test run and report back (hopefully today).

**Comment by rpetit3** [comment]
Hi @will-rowe 

I have tested @mjsull's latest release of Chromatiblock (v0.5.1) and everything looks good. The only major hold up on my end was the SVG, PNG, PDF output, which is now fixed in the latest version. I think @mjsull really went above and beyond responding to the rest of my comments.

I am happy to recommend this manuscript for publication!

Thank you for your time!
Robert

**Comment by will-rowe** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02451/joss.02451/10.21105.joss.02451.pdf)

**Comment by will-rowe** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1186/1471-2164-12-402 is OK
- 10.1093/bioinformatics/btq665 is OK
- 10.1101/gr.2289704 is OK
- 10.1186/1471-2164-13-202 is OK
- 10.1093/bioinformatics/btq413 is OK
- 10.1093/bioinformatics/bts217 is OK
- 10.1101/gr.092759.109 is OK
- 10.1101/548123 is OK
- 10.1093/bioinformatics/btr039 is OK

MISSING DOIs

- 10.1007/978-3-642-40453-5_17 may be a valid DOI for title: Sibelia: A Scalable and Comprehensive Synteny Block Generation Tool for Closely Related Microbial Genomes

INVALID DOIs

- None
```

**Comment by will-rowe** [comment]
Hi @mjsull - this is looking really nice. Just playing with the software now and am having no difficulties. The paper is also reading nicely. The only thing needing changing from my point of view is a fix for the Sibelia reference.

Thank you again @telatin and @rpetit3 - your reviews were excellent and this submission is looking great. Well done @mjsull with the responses.

@telatin has offered to take a look at your responses by end of the week - so we will wait on him to have a final look. Once has gives his approval - I'm happy for this to proceed to publication.

**Comment by telatin** [comment]
Hello @will-rowe!
@mjsull answered my comments and I have been happy with the improvements. 
I am happy to recommend this manuscript for publication!

Thank you all,
Andrea

**Comment by will-rowe** [comment]
perfect, thanks @telatin!

@mjsull - we can flag this for acceptance once you have done the following:
* fix that DOI
* tag a new release and archive it
* report the version and archive DOI back here

**Comment by mjsull** [comment]
Hi Will,

chromaitblock DOI has been added.

v1.0.0 released, the DOI: 10.5281/zenodo.4034604

**Comment by will-rowe** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1186/1471-2164-12-402 is OK
- 10.1093/bioinformatics/btq665 is OK
- 10.1101/gr.2289704 is OK
- 10.1186/1471-2164-13-202 is OK
- 10.1093/bioinformatics/btq413 is OK
- 10.1093/bioinformatics/bts217 is OK
- 10.1101/gr.092759.109 is OK
- 10.1101/548123 is OK
- 10.1007/978-3-642-40453-5_17 is OK
- 10.1093/bioinformatics/btr039 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by will-rowe** [comment]
perfect - thanks @mjsull

**Comment by will-rowe** [comment]
@whedon set 10.5281/zenodo.4034604 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4034604" target="_blank">10.5281/zenodo.4034604</a> is the archive.

**Comment by will-rowe** [comment]
@whedon set v1.0.0 as version

**Comment by whedon** [comment]
OK. v1.0.0 is the version.

**Comment by will-rowe** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1186/1471-2164-12-402 is OK
- 10.1093/bioinformatics/btq665 is OK
- 10.1101/gr.2289704 is OK
- 10.1186/1471-2164-13-202 is OK
- 10.1093/bioinformatics/btq413 is OK
- 10.1093/bioinformatics/bts217 is OK
- 10.1101/gr.092759.109 is OK
- 10.1101/548123 is OK
- 10.1007/978-3-642-40453-5_17 is OK
- 10.1093/bioinformatics/btr039 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by will-rowe** [comment]
This is with the EICs now. Thank you for a nice submission @mjsull and thank you very much @rpetit3 and @telatin for your reviews!

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1739

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1739, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by mjsull** [comment]
Awesome, thanks everyone.

**Comment by kyleniemeyer** [comment]
Hi @mjsull, I'm the EIC on duty this week, and doing some final checks before publishing. Could you address these issues?

- Please edit the metadata on the Zenodo archive so that the author names match what is in the paper. (By default Zenodo tends to just grab the GitHub usernames of contributors).
- Your paper is missing an explicit Statement of Need section, [required for all JOSS papers](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain), which should briefly illustrate the research purpose of the software. You can probably use some material given in the Introduction
- Also, can you break up the Introduction paragraph somewhat? Though this might be satisfied by fixing the previous issue
- please put commas after "i.e." throughout (I noticed mostly in Implementation section)

**Comment by mjsull** [comment]
Hi @kyleniemeyer,

All fixed.

Cheers,

Mitch

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02451/joss.02451/10.21105.joss.02451.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02451/joss.02451/10.21105.joss.02451.pdf) :page_facing_up: :point_left:

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1186/1471-2164-12-402 is OK
- 10.1093/bioinformatics/btq665 is OK
- 10.1101/gr.2289704 is OK
- 10.1186/1471-2164-13-202 is OK
- 10.1093/bioinformatics/btq413 is OK
- 10.1093/bioinformatics/bts217 is OK
- 10.1101/gr.092759.109 is OK
- 10.1101/548123 is OK
- 10.1007/978-3-642-40453-5_17 is OK
- 10.1093/bioinformatics/btr039 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1749

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1749, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1308918514410033153) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1750
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02451](https://doi.org/10.21105/joss.02451)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats  @mjsull on your article's publication in JOSS!

Many thanks to @telatin, @rpetit3, and @rpetit3 for reviewing this, and @will-rowe for editing it.

I see that the paper appears at https://joss.theoj.org/papers/10.21105/joss.02451, but the DOI does not yet resolve, so I'll keep this issue open until it does.

**Comment by rpetit3** [comment]
Congrats @mjsull!

@kyleniemeyer I'm not sure why but I'm on there twice (2x @rpetit3's) as a reviewer.

**Comment by kyleniemeyer** [comment]
@rpetit3 oh, that's strange, I totally missed that! 

@openjournals/dev can you fix this? @rpetit3 shows up twice as reviewer on the paper (likely due to showing up in the review issue twice—perhaps due to being added back at some point?)

**Comment by will-rowe** [comment]
Hi @kyleniemeyer and @openjournals/dev - this may have been my fault. For some reason, @rpetit3 was added to review as usual but had no access writes to the issue. We removed and added again and it fixed it, but not sure why he is listed twice! Sorry.

**Comment by arfon** [comment]
> @openjournals/dev can you fix this? @rpetit3 shows up twice as reviewer on the paper (likely due to showing up in the review issue twice—perhaps due to being added back at some point?)

This is now fixed but may take some time to update on the JOSS site due to caching.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02451/status.svg)](https://doi.org/10.21105/joss.02451)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02451">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02451/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02451/status.svg
   :target: https://doi.org/10.21105/joss.02451
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02451/status.svg)](https://doi.org/10.21105/joss.02451)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
